src/components/TenantAPISetupForm.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { FC, useState } from 'react';
+import { Alert, Button, Collapse, Container, Field, Form, HorizontalGroup, InfoBox, Input } from '@grafana/ui';
+import { DEFAULT_API_HOST } from './constants';
+
+interface FormValues {
+  adminApiToken: string;
+  apiHost?: string;
+}
+
+interface Props {
+  onSubmit: (values: FormValues) => void;
+  submissionError?: string;
+}
+
+const TenantAPISetupForm: FC<Props> = ({ onSubmit, submissionError }) => {
+  const [showAdvanced, setShowAdvanced] = useState(false);
+  return (
+    <Form
+      onSubmit={(values: FormValues) => onSubmit(values)}
```

Comment:
This creates a new callback function on each render, which can be often. `onSubmit={onSubmit}` is more efficient and will call with all arguments received (which isn't a problem here).

Reply:
I'm running into some collision of react-testing-library and react-hook-form. In the browser `onSubmit={onSubmit}` is just fine, but when running tests the `onSubmit` prop ends up getting passed the entire form element instead of the form values (but will work as expected when using the anonymous function) . I went ahead and changed it, but if you have any good ideas on why that might be I'm all ears!

Reply:
It looks like it passed in CI :shrug: